src/platform_impl/windows/util.rs

```diff
@@ -425,3 +426,72 @@ pub fn get_instance_handle() -> windows::Win32::Foundation::HMODULE {
 
   windows::Win32::Foundation::HMODULE(unsafe { &__ImageBase as *const _ as _ })
 }
+
+pub static WIN_VERSION: Lazy<windows_version::OsVersion> =
+  Lazy::new(windows_version::OsVersion::current);
+
+pub fn get_frame_thickness(dpi: u32) -> i32 {
+  let resize_frame_thickness = unsafe { GetSystemMetricsForDpi(SM_CXSIZEFRAME, dpi) };
+  let padding_thickness = unsafe { GetSystemMetricsForDpi(SM_CXPADDEDBORDER, dpi) };
+  resize_frame_thickness + padding_thickness
+}
+
+pub fn calculate_insets_for_dpi(dpi: u32) -> RECT {
+  // - On Windows 10
+  // The top inset must be zero, since if there is any nonclient area, Windows will draw
+  // a full native titlebar outside the client area. (This doesn't occur in the maximized
+  // case.)
+  //
+  // - On Windows 11
+  // The top inset is calculated using an empirical formula that I derived through various
+  // tests. Without this, the top 1-2 rows of pixels in our window would be obscured.
+
+  let frame_thickness = get_frame_thickness(dpi);
+
+  let top_inset = match WIN_VERSION.build {
+    v if v >= 2200 => (dpi as f32 / USER_DEFAULT_SCREEN_DPI as f32).round() as i32,
```

Comment:
```suggestion
    v if v >= 22000 => (dpi as f32 / USER_DEFAULT_SCREEN_DPI as f32).round() as i32,
```

Reply:
With this change, the window looks like this, the top border is missing and also not resizable from that edge

![image](https://github.com/user-attachments/assets/bd4345fe-4b65-43e9-ae60-5c65a47d8810)

I have read about an article https://kubyshkin.name/posts/win32-window-custom-title-bar-caption/ talking about this before, and the idea is that you need to draw your own top border to fake it...

Reply:
> With this change, the window looks like this, the top border is missing and also not resizable from that edge
> 
> I have read about an article https://kubyshkin.name/posts/win32-window-custom-title-bar-caption/ talking about this before, and the idea is that you need to draw your own top border to fake it...

I did read that one before but I don't want to add any custom drawing inside taol.

Reply:
I think this PR is good enough for aesthetics, don't you think?

As for, functionality maybe we bring back custom hittesting for only top border?

Reply:
> I did read that one before but I don't want to add any custom drawing inside taol.

Yeah, same thoughts here

> I think this PR is good enough for aesthetics, don't you think?

Yeah, it's good enough to me, at least it's much better than what we had before ;)

Also I have tested WindowsAppSdk sometime ago, and their `ExtendsContentIntoTitleBar` also had this problem (and they draw control buttons themselves as you tell can from the icons)

![image](https://github.com/user-attachments/assets/39152111-15fc-43f0-8d01-e43209c64777)

> As for, functionality maybe we bring back custom hittesting for only top border only on Windows 10?

I think we'll need this

Reply:
@Legend-Master does top-left and top-right resizing edge work and only top doesn't or all 3 don't work?

Reply:
actually looks like the issues exists in Windows 11 too.

Reply:
should be good to go now